circleguard/settings.py

```diff
@@ -181,9 +188,10 @@ def overwrite_outdated_settings():
         if last_version < version.parse(ver):
             for setting in changed_arr:
                 update_default(setting, DEFAULTS[TYPES[setting][1]][setting])
+    if last_version != version.parse(__version__) and last_version != version.parse(DEFAULTS["Core"]["last_version"]):
```

Comment:
very confused as to what this is doing. If someone downloads `v1.2.3` and previously had `v1.2.2`, `__version__ = 1.2.3` and `last_version = 1.2.2`. Why do we want to set `update_default` to `False` in that case? And why bother checking against the default `last_version`, which is just `0.0.0`?

Reply:
This is (was?) my solution to remove the message after the user updated.

The check for default is there because if a new user installs an old version, the update message would appear, but if the user quits and then opens it again the message would disappear (and only appear after an hour or so) cause the `last_version` changed from `0.0.0` to whatever version he is running.

Reply:
moved to update check logic